src/repr/src/scalar.rs

```diff
@@ -299,6 +299,69 @@ impl TryFrom<Datum<'_>> for Option<i64> {
     }
 }
 
+impl TryFrom<Datum<'_>> for u16 {
```

Comment:
Is there some sql stuff that automatically unlocks because of this?

Reply:
No. In order for stuff like `Datum::from(database_id),` to work I needed to implement `impl<'a> From<u64> for Datum<'a>`. I just figured while I was there I would implement the rest in both directions in case they would be useful later. I can remove this and the other ones that aren't currently used if you think that's cleaner.

Reply:
I find both arguments compelling (don't commit unused code; have useful code) so...whichever you prefer.

Reply:
I think I'll leave it then. It took me about 5 minutes to figure out I needed one of these so hopefully I save someone else 5+ minutes. Also every other type has all of these, so it seems consistent to add them for unsigned ints as well.